src/components/NavBar/SearchBarDropdown.tsx

```diff
@@ -111,15 +113,46 @@ export const SearchBarDropdown = ({
 }: SearchBarDropdownProps) => {
   const [hoveredIndex, setHoveredIndex] = useState<number | undefined>(0)
   const { history: searchHistory, updateItem: updateSearchHistory } = useSearchHistory()
-  const shortenedHistory = useMemo(() => searchHistory.slice(0, 2), [searchHistory])
+  const shortenedHistoryStalePrices = useMemo(() => searchHistory.slice(0, 2), [searchHistory])
+
+  // TODO(lynnshaoyu): possible to fetchSearchTokens by address instead of just by name?
+  const shortenedHistoryUpdatedPrices = useQueries(
+    shortenedHistoryStalePrices.map((tokenOrCollection) => ({
+      queryKey: ['tokenOrCollectionAddress', tokenOrCollection.address],
+      queryFn: () =>
+        isCollection(tokenOrCollection)
+          ? fetchSearchCollections(tokenOrCollection.address)
```

Comment:
These are not actually searches, as you know the specific address. You should ask the BE if there is a more appropriate endpoint, or if it is correct usage to re-use the existing endpoints with a fully specified address.